### PR TITLE
Feature/java 21

### PR DESCRIPTION
**Pull request by vaibhavcompanieshouse**

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: pom.xml [hunk 1]
```diff
@@ -53,19 +55,62 @@
...
+            <version>5.8.0</version>
             <scope>test</scope>
         </dependency>
+        <!--        <dependency>-->
+        <!--            <groupId>org.junit.vintage</groupId>-->
+        <!--            <artifactId>junit-vintage-engine</artifactId>-->
+        <!--            <version>${junit.vintage.engine.version}</version>-->
+        <!--            <scope>test</scope>-->
+        <!--        </dependency>-->
+        <!-- https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-engine -->
```
Comment (bcullerton) [inline pom.xml:103]: Can this be removed

File: pom.xml [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
         <surefire-plugin.version>2.22.2</surefire-plugin.version>
+        <junit.jupiter.version>5.8.0</junit.jupiter.version>
+        <jib-maven-plugin.version>3.4.0</jib-maven-plugin.version>
     </properties>
 
     <parent>
         <groupId>uk.gov.companieshouse</groupId>
         <artifactId>companies-house-parent</artifactId>
-        <version>1.3.0</version>
+        <version>2.1.2</version>
```
Comment (bcullerton) [inline pom.xml:30]: Can this be updated to 2.1.5

File: Makefile [hunk 3]
```diff
@@ -77,3 +77,13 @@ sonar:
...
 sonar-pr-analysis:
 	mvn sonar:sonar	-P sonar-pr-analysis
+
+
+.PHONY: security-check
+security-check:
+	mvn org.owasp:dependency-check-maven:update-only
+	mvn org.owasp:dependency-check-maven:check -DfailBuildOnCVSS=4 -DassemblyAnalyzerEnabled=false
+
+.PHONY: build-image
```
Comment (bcullerton) [inline Makefile:87]: Is this still needed?

File: src/main/java/uk/gov/companieshouse/extensions/api/config/HealthCheckController.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HealthCheckController {
+
+    @GetMapping("/extensions-api/healthcheck")
+    public ResponseEntity<String> healthcheck() {
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+}
```
Comment (bcullerton) [inline src/main/java/uk/gov/companieshouse/extensions/api/config/HealthCheckController.java:16]: Not a big deal but going forwards could we take a look at using spring-boot-actuator for healthchecks if they aren't present. Saves on having to write out the classes

File: src/main/java/uk/gov/companieshouse/extensions/api/requests/RequestsController.java [hunk 5]
```diff
@@ -1,29 +1,20 @@
...
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PatchMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-
+import org.springframework.web.bind.annotation.*;
```
Comment (bcullerton) [inline src/main/java/uk/gov/companieshouse/extensions/api/requests/RequestsController.java:7]: star import

File: src/test/java/uk/gov/companieshouse/extensions/api/attachments/AttachmentsServiceUnitTest.java [hunk 6]
```diff
@@ -46,8 +24,19 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/attachments/AttachmentsServiceUnitTest.java:36]: star imports here

File: src/test/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClientUnitTest.java [hunk 7]
```diff
@@ -1,39 +1,15 @@
...
-import org.springframework.http.MediaType;
-import org.springframework.http.ResponseEntity;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClientUnitTest.java:12]: stars here

File: src/test/java/uk/gov/companieshouse/extensions/api/authorization/CompanyAuthorizationInterceptorTest.java [hunk 8]
```diff
@@ -1,34 +1,23 @@
...
 import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
-
-import uk.gov.companieshouse.extensions.api.groups.Unit;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.extensions.api.logger.ApiLogger;
 
-@Category(Unit.class)
-@RunWith(MockitoJUnitRunner.class)
+import static org.mockito.Mockito.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/authorization/CompanyAuthorizationInterceptorTest.java:15]: star import

File: src/test/java/uk/gov/companieshouse/extensions/api/logger/ApiLoggerTest.java [hunk 9]
```diff
@@ -1,37 +1,33 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/logger/ApiLoggerTest.java:22]: star import

File: src/test/java/uk/gov/companieshouse/extensions/api/reasons/ReasonServiceUnitTest.java [hunk 10]
```diff
@@ -45,8 +28,17 @@
...
-@Category(Unit.class)
-@RunWith(MockitoJUnitRunner.class)
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static uk.gov.companieshouse.extensions.api.Utils.Utils.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/reasons/ReasonServiceUnitTest.java:38]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/reasons/ReasonsControllerUnitTest.java [hunk 11]
```diff
@@ -39,8 +20,17 @@
...
-@Category(Unit.class)
-@RunWith(MockitoJUnitRunner.class)
+import java.util.List;
+import java.util.Objects;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.companieshouse.extensions.api.Utils.Utils.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/reasons/ReasonsControllerUnitTest.java:30]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/requests/MongoDBTest.java [hunk 12]
```diff
@@ -9,17 +9,13 @@
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Ignore;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/requests/MongoDBTest.java:12]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestControllerUnitTest.java [hunk 13]
```diff
@@ -27,27 +25,15 @@
...
-import uk.gov.companieshouse.extensions.api.logger.ApiLogger;
-import uk.gov.companieshouse.extensions.api.reasons.ExtensionReasonEntity;
-import uk.gov.companieshouse.extensions.api.response.ListResponse;
-import uk.gov.companieshouse.service.ServiceException;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.companieshouse.extensions.api.Utils.Utils.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestControllerUnitTest.java:33]: .* imports

File: src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestMapperUnitTest.java [hunk 14]
```diff
@@ -1,21 +1,19 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.companieshouse.extensions.api.reasons.ExtensionReasonEntity;
 import uk.gov.companieshouse.service.links.Links;
 
-@Category(Unit.class)
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static uk.gov.companieshouse.extensions.api.Utils.Utils.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestMapperUnitTest.java:13]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestRepositoryIntegrationTest.java [hunk 15]
```diff
@@ -1,34 +1,40 @@
...
-@Category(Integration.class)
-@RunWith(SpringRunner.class)
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestRepositoryIntegrationTest.java:24]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestServiceUnitTest.java [hunk 16]
```diff
@@ -1,39 +1,31 @@
...
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import static uk.gov.companieshouse.extensions.api.Utils.Utils.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/requests/RequestServiceUnitTest.java:25]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/attachments/AttachmentsControllerUnitTest.java [hunk 17]
```diff
@@ -13,28 +15,19 @@
...
+import java.io.IOException;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/attachments/AttachmentsControllerUnitTest.java:25]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferGatewayIntegrationTest.java [hunk 18]
```diff
@@ -1,54 +1,44 @@
...
+import java.nio.file.Files;
+import java.util.Objects;
+import java.util.concurrent.Callable;
 
 import static org.awaitility.Awaitility.await;
 import static org.hamcrest.Matchers.equalTo;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferGatewayIntegrationTest.java:35]: .* import

File: src/test/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferGatewayMockIntegrationTest.java [hunk 19]
```diff
@@ -1,32 +1,38 @@
...
 
-import org.junit.AfterClass;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.rules.ExpectedException;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.*;
```
Comment (bcullerton) [inline src/test/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferGatewayMockIntegrationTest.java:3]: .* import

**Review by bcullerton — CHANGES_REQUESTED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by bcullerton — APPROVED** [review]

**Pull request merged**